DatadogFlags/Sources/FlagsTypes.swift

```diff
@@ -23,7 +23,73 @@ public enum FlagsError: Error {
     case unsupportedSite(String)
 }
 
-internal struct FlagsMetadata {
+// TODO: FFL-1016 Include everything to fully hydrate a client (evaluationContext, clientConfiguration, anything else?)
+internal struct FlagsMetadata: Codable {
     let fetchedAt: Double // Timestamp in milliseconds (JavaScript-style)
     let context: FlagsEvaluationContext?
 }
+
+/// A Codable wrapper for flags dictionary to enable JSON serialization in DataStore
+internal struct CodableFlags: Codable {
+    let flags: [String: CodableAny]
+
+    init(flags: [String: Any]) {
+        self.flags = flags.compactMapValues { CodableAny($0) }
+    }
+
+    func toDictionary() -> [String: Any] {
+        return flags.compactMapValues { $0.value }
+    }
+}
+
+/// A type-erased Codable wrapper to handle Any values in flags
+internal struct CodableAny: Codable {
```

Comment:
question/ Doesn't it duplicate the existing [`DatadogInternal/AnyCodable`](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/Codable/AnyCodable.swift)?